include/dlaf/mc/cholesky.h

```diff
@@ -0,0 +1,306 @@
+
+          // Avoid useless communications if one-row communicator grid
+          if (row_comm_size > 1) {
+            // Broadcast the panel row-wise
+            hpx::dataflow(hpx::util::unwrapping([](auto&& tile, auto&& comm_wrapper) {
+                            dlaf::comm::sync::broadcast::send(comm_wrapper().rowCommunicator(), tile);
+                          }),
+                          mat.read(LocalTileIndex{i_local, k_local_col}), serial_comm());
+          }
+
+          panel[i_local] = mat.read(LocalTileIndex{i_local, k_local_col});
+        }
+      }
+      else {
+        for (SizeType i_local = distr.nextLocalTileFromGlobalTile<Coord::Row>(k + 1);
+             i_local < localnrtile_rows; ++i_local) {
+          auto i = distr.globalTileFromLocalTile<Coord::Row>(i_local);
+
+          // Avoid useless communications if one-row communicator grid
+          if (row_comm_size > 1) {
+            // Receiving the panel
+            panel[i_local] = hpx::dataflow(  //
+                hpx::util::unwrapping(
+                    [](auto index, auto&& tile_size, auto&& comm_wrapper) -> Tile<const T, Device::CPU> {
+                      memory::MemoryView<T, Device::CPU> mem_view(
+                          util::size_t::mul(tile_size.rows(), tile_size.cols()));
+                      Tile<T, Device::CPU> tile(tile_size, std::move(mem_view), tile_size.rows());
+                      dlaf::comm::sync::broadcast::receive_from(index, comm_wrapper().rowCommunicator(),
+                                                                tile);
+                      return std::move(tile);
+                    }),
+                k_rank_col, mat.tileSize(GlobalTileIndex(i, k)), serial_comm());
+          }
+        }
+      }
+
+      auto nextlocaltilek = distr.nextLocalTileFromGlobalTile<Coord::Col>(k + 1);
+      for (SizeType j_local = nextlocaltilek; j_local < localnrtile_cols; ++j_local) {
+        auto j = distr.globalTileFromLocalTile<Coord::Col>(j_local);
+
+        // Choose "high priority" for first tile of the trailing matrix
+        // Version 1: only the global k + 1 column is "high priority"
+        auto trailing_matrix_executor = (j == k + 1) ? matrix_HP_executor : matrix_normal_executor;
+
+        // TODO: benchmark to find faster
+        // Version 2: the next column of each rank is "high priority"
+        //        auto trailing_matrix_executor = (j_local == nextlocaltilek) ? matrix_HP_executor :
+        //        matrix_normal_executor;
+
+        hpx::shared_future<Tile<const T, Device::CPU>> col_panel;
+
+        auto j_rank_row = distr.rankGlobalTile<Coord::Row>(j);
+
+        if (mat.rankIndex().row() == j_rank_row) {
+          auto i_local = distr.localTileFromGlobalTile<Coord::Row>(j);
+
+          // Avoid useless communications if one-row communicator grid and if on the last panel
+          if (col_comm_size > 1 && j != (mat.size().cols() - 1)) {
```

Comment:
Look similar previous comment.

Reply:
@albestro This comment is unclear, try to be more specific in these cases... (especially if the comments are from multiple reviews). Are you referring to the comment of line 165?

Reply:
Yes, exactly.

Sorry, but I usually don't care about different reviews for the same PR, I just look at the code and I see all available comments. Got it.